libraries/AP_Airspeed/AP_Airspeed_SDP3X.cpp

```diff
@@ -0,0 +1,207 @@
+            continue;
+
+        // lots of retries during probe
+        _dev->set_retries(10); 
+
+	// step 1 - reset on broadcast
+        uint8_t reset_cmd = SDP3X_RESET_CMD;
+        ret = _dev->transfer((uint8_t *)&reset_cmd, 1, nullptr, 0);
+        if (!ret)
+            printf("SDP3X: reset command failed\n");
+
+	// waiting 20ms to reset
+	hal.scheduler->delay_microseconds(20000);
+
+	// step 2 - configure
+        uint8_t cmd[2] = {(SDP3X_CONT_MEAS_AVG_MODE & 0xFF00) >> 8, SDP3X_CONT_MEAS_AVG_MODE & 0xFF};
+        ret = _dev->transfer((uint8_t *)&cmd, 2, nullptr, 0);
+        if (!ret)
+            printf("SDP3X: continued measure command failed\n");
+
+	// waiting again
+	hal.scheduler->delay_microseconds(20000);
+
+	// step 3 - get scale
+	uint8_t val[9];
+	ret = _dev->transfer(nullptr, 0, &val[0], sizeof(val));    
+        if (!ret)
+            printf("SDP3X: get scale value failed\n");
+
+    	// Check the CRC
+        if (!_crc(&val[0], 2, val[2]) || !_crc(&val[3], 2, val[5]) || !_crc(&val[6], 2, val[8]))
+	    printf("SDP3X: CRC failed !\n");
+
+	_scale = (((uint16_t)val[6]) << 8) | val[7];
+	printf("Scale value is : %d\n", _scale);
+        _collect();
+        _dev->get_semaphore()->give();
+        found = (_last_sample_time_ms != 0);
+        if (found)
+        {
+	    switch (_scale)
+	    {
+		case SDP3X_SCALE_PRESSURE_SDP31:
+            		printf("SDP31: Found sensor on bus %u address 0x%02x\n",
+				addresses[i].bus, addresses[i].addr);
+			break;
+		case SDP3X_SCALE_PRESSURE_SDP32:
+		        printf("SDP32: Found sensor on bus %u address 0x%02x\n",
+				addresses[i].bus, addresses[i].addr);
+			break;
+		case SDP3X_SCALE_PRESSURE_SDP33:
+            		printf("SDP33: Found sensor on bus %u address 0x%02x\n",
+				addresses[i].bus, addresses[i].addr);
+			break;
+	    }
+            break;
+        }
+    }
+    if (!found)
+    {
+        printf("SDP3X: no sensor found\n");
+        return false;
+    }
+
+    // drop to 2 retries for runtime
+    _dev->set_retries(2);
+    
+    _dev->register_periodic_callback(50000,
+                                     FUNCTOR_BIND_MEMBER(&AP_Airspeed_SDP3X::_timer, void));
+    return true;
+}
+
+// read the values from the sensor
+void AP_Airspeed_SDP3X::_collect()
+{
+	// read 9 bytes from the sensor
+	uint8_t val[6];
+	int ret = _dev->transfer(nullptr, 0, &val[0], sizeof(val));
+	if (!ret)
+		printf("SDP3X: Collect error\n");
+	// Check the CRC
+	if (!_crc(&val[0], 2, val[2]) || !_crc(&val[3], 2, val[5]))
+	    printf("Error, BAD CRC collected\n");
+
+	int16_t P = (((int16_t)val[0]) << 8) | val[1];
+	int16_t temp = (((int16_t)val[3]) << 8) | val[4];
+
+	float P_float = (float)P;
+	float temp_float = (float)temp;
+
+	float scale_float = (float)_scale;
+
+	float diff_press_pa = P_float / scale_float;
+	float temperature = temp_float / (float)SDP3X_SCALE_TEMPERATURE;
+
+        if (sem->take(HAL_SEMAPHORE_BLOCK_FOREVER))
```

Comment:
I don't like taking forever, but I'll defer to @tridge on it. But the main issue is that the getters for pressure and temperature aren't using the semaphore.